src/charm.py

```diff
@@ -119,14 +115,19 @@ def __init__(self, *args):
             ),
         )
 
-        self.grafana_dashboard_provider = GrafanaDashboardProvider(
-            charm=self, dashboards_path=self._grafana_dashboards_path
+        self.grafana_dashboard_provider = GrafanaDashboardProvider(self)
+        self.grafana_dashboard_provider_devices = GrafanaDashboardProvider(
+            self,
+            relation_name="grafana-dashboard-devices",
+            dashboards_path="src/grafana_dashboards/devices",
```

Comment:
Maybe we should add a comment mentioning that this path doesn't exist but is required by the constructor.

Reply:
Technically the path exists, otherwise `GrafanaDashboardProvider` complains...

Reply:
and it complains as a warning